back/engines/commercial/id_oostende_rrn/app/lib/id_oostende_rrn/oostende_rrn_verification.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+module IdOostendeRrn
+  class OostendeRrnVerification
+    include Verification::VerificationMethod
+
+    def verification_method_type
+      :manual_sync
+    end
+
+    def id
+      '00c4faa6-4b0f-11ed-b878-0242ac120002'
+    end
+
+    def name
+      'oostende_rrn'
+    end
+
+    def config_parameters
+      %i[api_key environment]
+    end
+
+    def config_parameters_schema
+      {
+        environment: {
+          type: 'string',
+          enum: %w[dv qa production],
+          private: true
+        }
+      }
+    end
+
+    def locked_custom_fields
+      [config[:custom_field_key]]
+    end
+
+    def verification_parameters
+      [:rrn]
+    end
+
+    def verify_sync(rrn:)
+      cleaned_rrn = clean_rrn(rrn)
+      raise Verification::VerificationService::ParameterInvalidError, 'rrn' unless rrn_valid?(cleaned_rrn)
+
+      validate_citizen!(cleaned_rrn)
+    end
+
+    private
+
+    def validate_citizen!(rrn)
+      api = WijkBudgetApi.new(api_key: config[:api_key], environment: config[:environment])
+      response = api.verificatie(rrn)
+
+      raise RuntimeError(response) unless response.success?
+
+      body = response.parsed_response
+      reason = body.dig('verificatieResultaat', 'redenNietGeldig')
+
+      raise Verification::VerificationService::NoMatchError if reason&.include? 'ERR10'
+      # raise Verification::VerificationService::NotEntitledError, 'lives_outside' if reason&.include? 'ERR11'
+      raise Verification::VerificationService::NotEntitledError, 'too_young' if reason&.include? 'ERR12'
+
+      raise Verification::VerificationService::NoMatchError unless body.dig('verificatieResultaat', 'geldig')
+
+      { uid: rrn }.compact
```

Comment:
In `rrn_valid?()` we already check if `rnn` is empty, so I think we can remove the `compact` call on the hash here.